keras_trainer/trainer.py

```diff
@@ -85,44 +90,53 @@ def run(self):
-            }
-            model_kwargs.update(self.model_kwargs)
-            model = self.model_spec.klass(**model_kwargs)
-
-            # Expand the base model to match the spec target_size.
-            model_kwargs.update({
-                'input_shape': self.model_spec.target_size,
-                'weights': None
-            })
-            expanded_model = self.model_spec.klass(**model_kwargs)
-            for i in range(1, len(expanded_model.layers)):
-                expanded_model.layers[i].set_weights(model.layers[i].get_weights())
-            model = expanded_model
+        if self.checkpoint_path is not None:
+            model = load_model(self.checkpoint_path)
         else:
-            model = self.model_spec.klass(
-                input_shape=self.model_spec.target_size,
-                weights=self.weights,
-                include_top=False,
-                pooling=self.pooling
-            )
-
-        # Include the given top layers.
-        if self.top_layers is None:
-            layer = Dense(self.num_classes, name='dense')(model.output)
-            self.top_layers = Activation('softmax', name='act_softmax')(layer)
-        model = Model(model.input, self.top_layers)
+            if self.model_spec.klass == MobileNet:
+                # Initialize the base with valid target_size.
+                model_kwargs = {
+                    'input_shape': (224, 224, 3),
+                    'weights': self.weights,
+                    'include_top': False,
+                    'pooling': self.pooling
+                }
+                model_kwargs.update(self.model_kwargs)
+                model = self.model_spec.klass(**model_kwargs)
+
+                # Expand the base model to match the spec target_size.
+                model_kwargs.update({
+                    'input_shape': self.model_spec.target_size,
+                    'weights': None
+                })
+                expanded_model = self.model_spec.klass(**model_kwargs)
+                for i in range(1, len(expanded_model.layers)):
+                    expanded_model.layers[i].set_weights(model.layers[i].get_weights())
+                model = expanded_model
+            else:
+                model = self.model_spec.klass(
+                    input_shape=self.model_spec.target_size,
+                    weights=self.weights,
+                    include_top=False,
+                    pooling=self.pooling
+                )
+
+            # Include the given top layers.
+            if self.top_layers is None:
+                layer = Dense(self.num_classes, name='dense')(model.output)
+                self.top_layers = Activation('softmax', name='act_softmax')(layer)
+            model = Model(model.input, self.top_layers)
 
         # Print the model summary.
         if self.verbose:
             model.summary()
 
+        # GPU multiprocessing (if None we use all available GPUs)
+        if self.num_gpu is None:
```

Comment:
Can you move this check/assignment to the constructor around line 57 where we do the `num_classes` validation? Something like this:
```python
if self.num_gpus is None:
    self.num_gpus = detect_num_gpus()
```